# global robot configuration and details

robot_config:
    joint_names:
      - panda_joint1
      - panda_joint2
      - panda_joint3
      - panda_joint4
      - panda_joint5
      - panda_joint6
      - panda_joint7
    arm_id: panda
        # Activate rate limiter? [true|false]
    rate_limiting: true
    # Cutoff frequency of the low-pass filter. Set to >= 1000 to deactivate.
    cutoff_frequency: 100
    # Internal controller for motion generators [joint_impedance|cartesian_impedance]
    internal_controller: joint_impedance

    neutral_pose:
        panda_joint1: -0.017792060227770554 
        panda_joint2: -0.7601235411041661  
        panda_joint3: 0.019782607023391807 
        panda_joint4: -2.342050140544315 
        panda_joint5: 0.029840531355804868 
        panda_joint6: 1.5411935298621688 
        panda_joint7: 0.7534486589746342 

    joint_config:
        joint_acceleration_limit:
            panda_joint1: 15.0 # rad / sec^2
            panda_joint2: 7.5  # rad / sec^2
            panda_joint3: 10.0 # rad / sec^2
            panda_joint4: 12.5 # rad / sec^2
            panda_joint5: 15.0 # rad / sec^2
            panda_joint6: 20.0 # rad / sec^2
            panda_joint7: 20.0 # rad / sec^2
            # panda_finger_joint1: 12.0 # rad / sec^2
            # panda_finger_joint2: 12.0 # rad / sec^2

        joint_velocity_limit:
            panda_joint1: 2.1750 # rad / sec
            panda_joint2: 2.1750 # rad / sec
            panda_joint3: 2.1750 # rad / sec
            panda_joint4: 2.1750 # rad / sec
            panda_joint5: 2.6100 # rad / sec
            panda_joint6: 2.6100 # rad / sec
            panda_joint7: 2.6100 # rad / sec
            # panda_finger_joint1: 2.0 # rad / sec
            # panda_finger_joint2: 2.0 # rad / sec

        joint_position_limit:
            lower: 
                panda_joint1: -2.8973 # rad
                panda_joint2: -1.7628 # rad
                panda_joint3: -2.8973 # rad
                panda_joint4: -3.0718 # rad
                panda_joint5: -2.8973 # rad
                panda_joint6: -0.0175 # rad
                panda_joint7: -2.8973 # rad
            upper:
                panda_joint1: 2.8973 # rad
                panda_joint2: 1.7628 # rad
                panda_joint3: 2.8973 # rad
                panda_joint4: -0.0698 # rad
                panda_joint5: 2.8973 # rad
                panda_joint6: 3.7525 # rad
                panda_joint7: 2.8973 # rad

        joint_effort_limit:
            panda_joint1: 87 # Nm
            panda_joint2: 87 # Nm
            panda_joint3: 87 # Nm
            panda_joint4: 87 # Nm
            panda_joint5: 12 # Nm
            panda_joint6: 12 # Nm
            panda_joint7: 12 # Nm            

gripper_config:
  joint_names:
    - panda_finger_joint1
    - panda_finger_joint2
  default_speed: 0.1  # [m/s]
  default_grasp_epsilon:
    inner: 0.005 # [m]
    outer: 0.005 # [m]

controllers_config:
    position_controller: "franka_position_servo/joint_position_controller"
    trajectory_controller: "position_joint_trajectory_controller"
